vNext/UnityGLTFv2.cs

```diff
@@ -1,63 +1,70 @@
         );
 
     /// <summary>
     /// Creates a Unity GameObject from a glTF scene
     /// </summary>
-    /// <param name="unityGLTFFObject">Object which contains information to parse</param>
+    /// <param name="unityGLTFObject">Object which contains information to parse</param>
     /// <param name="sceneId">Scene of the glTF to load</param>
+    /// <param name="progress">Progress of load completion</param>
     /// <returns>The created Unity object</returns>
-    public Task<GameObject> ImportSceneAsync(
-        UnityGLTFObject unityGLTFFObject,
-        int sceneId = -1
+    public virtual Task<GameObject> ImportSceneAsync(
+        UnityGLTFObject unityGLTFObject,
+        int sceneId = -1,
+        IProgress<int> progress = null,
+        CancellationToken cancellationToken = CancellationToken.None
         );
 
     /// <summary>
     /// Creates a Unity GameObject from a glTF node
     /// </summary>
-    /// <param name="unityGLTFFObject">Object which contains information to parse</param>
+    /// <param name="unityGLTFObject">Object which contains information to parse</param>
     /// <param name="nodeId">Node of the glTF object to load.</param>        
     /// <returns>The created Unity object</returns>
-    public Task<GameObject> ImportNodeAsync(
-        UnityGLTFObject unityGLTFFObject,
-        int nodeId
+    public virtual Task<GameObject> ImportNodeAsync(
+        UnityGLTFObject unityGLTFObject,
+        int nodeId,
+        CancellationToken cancellationToken = CancellationToken.None
         );
 
     /// <summary>
     /// Creates a Unity Texture2D from a glTF texture
     /// </summary>
-    /// <param name="unityGLTFFObject">Object which contains information to parse</param>
+    /// <param name="unityGLTFObject">Object which contains information to parse</param>
     /// <param name="textureId">Texture to load from glTF object.</param>        
     /// <returns>The created Unity object</returns>
-    public Task<Texture2D> ImportTextureAsync(
-        UnityGLTFObject unityGLTFFObject,
-        int textureId
+    public virtual Task<Texture2D> ImportTextureAsync(
+        UnityGLTFObject unityGLTFObject,
+        int textureId,
+        CancellationToken cancellationToken = CancellationToken.None,
         );
 }
 
 // UnityNode.cs
-public partial class UnityGLTFImporter
+public virtual partial class Importer
 {
-    private Task<GameObject> ConstructNode();
+    private virtual Task<GameObject> ConstructNode();
 }
 
 // UnityTexture.cs
-public partial class UnityGLTFImporter
+public virtual partial class Importer
 {
-    private Task<Texture2D> ConstructTexture();
+    private virtual Task<Texture2D> ConstructTexture();
 }
 
-public class UnityGLTFImporterConfig
+public class ImporterConfig
 {
-    public UnityGLTFImporterConfig();
-    public UnityGLTFImporterConfig(GLTFExtensionRegistry registry, GLTFImportOptions importOptions);
+    public ImporterConfig();

```

Comment:
Why do we need this paramterless constructor?

Reply:
The default constructor is used for when there are no options that are being set

Reply:
Why not just pass in null if you don't want to set options? It seems weird that the options are to either provide extensions *and* options, *or* nothing.